lib/panomosity/generalized_neighborhood.rb

```diff
@@ -88,6 +88,17 @@ def similar_neighborhoods(type: :horizontal)
         type == :horizontal ? @horizontal_similar_neighborhoods : @vertical_similar_neighborhoods
       end
 
+      def similar_neighborhoods!(type: :horizontal)
```

Comment:
Let's rename this method to include an action. I like the `!` but I think that something like `fetch_similar_neighborhoods!` or `(get|calculate)_similar_neighborhoods!` will make it more clear that this can raise an error.

Reply:
I think it should keep the same name as the original since it does the same thing.
Should I rename the original `fetch_similar_neighborhoods` method as well?
It is used quite a bit.

Reply:
Yeah honestly it's better to keep it this way. Rereading it now it makes more sense since you expect bangs to return errors. This will need to be refactored later likely into a `SimilarNeighborhood` class or something similar.